pkg/promlib/converter/prom.go

```diff
@@ -857,14 +858,15 @@ func readHistogram(iter *sdkjsoniter.Iterator, hist *histogramInfo) error {
 }
 
 func appendValueFromString(iter *sdkjsoniter.Iterator, field *data.Field) error {
-	var err error
-	var s string
-	if s, err = iter.ReadString(); err != nil {
+	// Read the string directly into our buffer
+	buf, err := iter.ReadStringAsSlice()
+	if err != nil {
 		return err
 	}
 
-	var v float64
-	if v, err = strconv.ParseFloat(s, 64); err != nil {
+	// Convert to float64 without allocating a new string
+	v, err := strconv.ParseFloat(*(*string)(unsafe.Pointer(&buf)), 64)
```

Comment:
https://github.com/search?q=org%3Agrafana+yoloString&type=code

Reply:
I'd love to have this reference or comment about this in code as well.

Reply:
i decided to move this whole change into its own PR to make it granular

Reply:
Here is the PR https://github.com/grafana/grafana/pull/102024